--- a/rogue/command.c	2025-08-16 17:49:39
+++ b/rogue/command.c	2025-08-23 15:39:14
@@ -291,7 +291,7 @@
 		    wrefresh(curscr);
 		when 'v':
 		    after = FALSE;
-		    msg("version %s. (mctesq was here)", release);
+		    msg("version %s. (mctesq was here) PicoCalc %s.", release, PICOCALC_ROGUE_VERSION);
 		when 'S': 
 		    after = FALSE;
 		    save_game();
--- a/rogue/extern.c	2025-08-18 21:28:34
+++ b/rogue/extern.c	2025-08-18 21:22:05
@@ -348,14 +348,14 @@
     {CTRL('U'),	"	run up & right until adjacent",		FALSE},
     {CTRL('B'),	"	run down & left until adjacent",	FALSE},
     {CTRL('N'),	"	run down & right until adjacent",	FALSE},
-    {'\0',	"	<SHIFT><dir>: run that way",		TRUE},
-    {'\0',	"	<CTRL><dir>: run till adjacent",	TRUE},
-    {'f',	"<dir>	fight till death or near death",	TRUE},
+    {'\0',	"	<SHFT><dir>: run",		TRUE},
+    {'\0',	"	<CTRL><dir>: run adj",	TRUE},
+    {'f',	"<dir>	fight till (near) death",	TRUE},
     {'t',	"<dir>	throw something",			TRUE},
-    {'m',	"<dir>	move onto without picking up",		TRUE},
-    {'z',	"<dir>	zap a wand in a direction",		TRUE},
+    {'m',	"<dir>	move without picking up",		TRUE},
+    {'z',	"<dir>	zap a wand",		TRUE},
     {'^',	"<dir>	identify trap type",			TRUE},
-    {'s',	"	search for trap/secret door",		TRUE},
+    {'s',	"	search for trap/secret",		TRUE},
     {'>',	"	go down a staircase",			TRUE},
     {'<',	"	go up a staircase",			TRUE},
     {'.',	"	rest for a turn",			TRUE},
@@ -377,7 +377,7 @@
     {']',	"	print current armor",			TRUE},
     {'=',	"	print current rings",			TRUE},
     {'@',	"	print current stats",			TRUE},
-    {'D',	"	recall what's been discovered",		TRUE},
+    {'D',	"	recall discovered",		TRUE},
     {'o',	"	examine/set options",			TRUE},
     {CTRL('R'),	"	redraw screen",				TRUE},
     {CTRL('P'),	"	repeat last message",			TRUE},
@@ -385,7 +385,7 @@
     {'S',	"	save game",				TRUE},
     {'Q',	"	quit",					TRUE},
     {'!',	"	shell escape",				TRUE},
-    {'F',	"<dir>	fight till either of you dies",		TRUE},
+    {'F',	"<dir>	fight till one dies",		TRUE},
     {'v',	"	print version number",			TRUE},
     {0,		NULL }
 };
--- a/rogue/io.c	2025-08-18 21:28:38
+++ b/rogue/io.c	2025-08-18 21:22:56
@@ -220,7 +220,7 @@
     if (stat_msg)
     {
 	move(0, 0);
-        msg("Level: %d  Gold: %-5d  Hp: %*d(%*d)  Str: %2d(%d)  Arm: %-2d  Exp: %d/%ld  %s",
+        msg("Lvl: %d G: %-5d H: %*d(%*d) S: %2d(%d) A: %-2d E: %d/%ld %s",
 	    level, purse, hpwidth, pstats.s_hpt, hpwidth, max_hp, pstats.s_str,
 	    max_stats.s_str, 10 - s_arm, pstats.s_lvl, pstats.s_exp,
 	    state_name[hungry_state]);
@@ -229,7 +229,7 @@
     {
 	move(STATLINE, 0);
                 
-        printw("Level: %d  Gold: %-5d  Hp: %*d(%*d)  Str: %2d(%d)  Arm: %-2d  Exp: %d/%d  %s",
+        printw("Lvl: %d G: %-5d H: %*d(%*d) S: %2d(%d) A: %-2d E: %d/%d %s",
 	    level, purse, hpwidth, pstats.s_hpt, hpwidth, max_hp, pstats.s_str,
 	    max_stats.s_str, 10 - s_arm, pstats.s_lvl, pstats.s_exp,
 	    state_name[hungry_state]);
--- a/rogue/main.c	2025-08-23 15:54:36
+++ b/rogue/main.c	2025-08-23 15:54:42
@@ -13,6 +13,7 @@
 #include <signal.h>
 #include <time.h>
 #include <curses.h>
+#include "pico/rand.h"
 #include "rogue.h"
 
 /*
@@ -20,7 +21,7 @@
  *	The main program, of course
  */
 int
-main(int argc, char **argv, char **envp)
+rogue_main(int argc, char **argv, char **envp)
 {
     char *env;
     int lowtime;
@@ -45,11 +46,8 @@
     /*
      * get home and options from environment
      */
-
-    strncpy(home, md_gethomedir(), MAXSTR);
 
-    strcpy(file_name, home);
-    strcat(file_name, "rogue.save");
+    strcpy(file_name, "/Rogue/rogue.save");
 
     if ((env = getenv("ROGUEOPTS")) != NULL)
 	parse_opts(env);
@@ -185,7 +183,7 @@
 int
 rnd(int range)
 {
-    return range == 0 ? 0 : abs((int) RN) % range;
+    return range == 0 ? 0 : get_rand_32() % range;
 }
 
 /*
--- a/rogue/rogue.h	2025-08-18 21:28:51
+++ b/rogue/rogue.h	2025-08-23 15:39:37
@@ -11,6 +11,7 @@
  */
 
 #include "extern.h"
+#include "version.h"
 
 #undef lines 
 
@@ -27,8 +28,8 @@
 #define AMULETLEVEL	26
 #define	NUMTHINGS	7	/* number of types of things */
 #define MAXPASS		13	/* upper limit on number of passages */
-#define	NUMLINES	24
-#define	NUMCOLS		80
+#define	NUMLINES	32
+#define	NUMCOLS		64
 #define STATLINE		(NUMLINES - 1)
 #define BORE_LEVEL	50
 
--- a/rogue/save.c	2025-08-16 17:49:39
+++ b/rogue/save.c	2025-08-23 16:14:28
@@ -19,6 +19,8 @@
 #include <curses.h>
 #include "rogue.h"
 #include "score.h"
+
+extern void rogue_exit(int status);
 
 typedef struct stat STAT;
 
@@ -151,7 +153,7 @@
     rs_save_file(savef);
     fflush(savef);
     fclose(savef);
-    exit(0);
+    rogue_exit(0);
 }
 
 /*
